src/_pytest/nodes.py

```diff
@@ -515,6 +515,7 @@ def __init__(
         session: Optional["Session"] = None,
         nodeid: Optional[str] = None,
     ) -> None:
+        fspath = py.path.local(os.path.abspath(str(fspath)))
```

Comment:
`py.path.local` already runs `abspath` on its input:

https://github.com/pytest-dev/py/blob/2da2caea38812eaa3ce09dd5292e3635ce9b16c8/py/_path/local.py#L153-L163

So I wonder how this line has any effect. I am assuming that `abspath` is idempotent, but maybe it isn't on Windows? Or maybe the troublesome path comes from the `os.getcwd()` code path, and the result of that isn't absolute?

Basically I'm interested to know a bit more what is going on here:

1. Where does the input `fspath` come from, and what string was used to initialize it
2. How is it transformed by this line

Reply:
Good questions, I didn't realize that `py.path.local` would call `abspath` on its input.

Digging further, I found the place where the path is not correctly realized:

https://github.com/pytest-dev/pytest/blob/3f0abcc6a5e0ddf1c17d42125421c6816fcf3178/src/_pytest/main.py#L696-L697

```python
>>> import py.path
>>> py.path.local(r'd:\projects').join(r'\foo', abs=True)
local('\\foo')
```

So it is a problem with `py.path.local.join()`.

Decided to throw out that bit of code and use the `Path` API locally, which doesn't have that bug.

Thanks for the inquiry, that lead me to investigate it further and find the real cause of the issue. 👍 😅